Add flag to keep table at application launch

tedyu opened the pull request:
Currently applications such as SqlInserts and SqlSecondaryIndex do not drop existing table in createTablesIfNeeded().

This PR changes the behavior where by default, the table would be truncated in createTablesIfNeeded().

A flag, keep_table (without argument), is added. When the flag is specified, the table would be kept (and not truncated).

This is to facilitate simultaneous runs of the same application against the same DB table.

tedyu commented:
@rkarthik007 @m-iancu 
Please review.

kmuthukk commented:
@tedyu --

Can we make "truncate" table the default behavior for SQL* workloads in sample apps, unless --keep_table or --drop_table_name options are specified?

tedyu commented:
I have updated the PR where table is truncated if neither --keep_table nor --drop_table_name option is specified.

tedyu commented:
ping @rkarthik007 @m-iancu for review

kmuthukk commented on src/main/java/com/yugabyte/sample/apps/AppConfig.java:
```diff
@@ -125,8 +127,8 @@
-  // Does the table need to be dropped.
-  public boolean shouldDropTable = false;
+  // should we drop / truncate table
+  public AppBase.TableOp tableOp = TableOp.TruncateTable;
```
@tedyu -- my understanding was that we wanted the truncate behavior only for YSQL (because with SQL, simply rerunning the tests will result in duplicate key exceptions). With YCQL, it has UPSERT semantics by default -- so let's us not change the default behavior for YCQL. We could have the --truncate option, but perhaps not make it the default.

kmuthukk commented on src/main/java/com/yugabyte/sample/apps/AppConfig.java:
```diff
@@ -125,8 +127,8 @@
 
-  // Does the table need to be dropped.
-  public boolean shouldDropTable = false;
+  // should we drop / truncate table
```
Comments begin with upper case, and end with period (as per our coding convention).

tedyu commented:
Moved "keep_table" checking into initializeNumKeys() so that it is easy to observe the new logic.

Added @Override to createTablesIfNeeded for MultiTableMultiIndexInserts.java

kmuthukk reviewed (approved):
Thx @tedyu !
